### PR TITLE
Adds debug logging if specs mismatch in findCRD

### PR DESCRIPTION
**Pull request by jmprusi**
Helpful for debugging #297

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -934,6 +936,7 @@ func findCRD(crdName string, crds []*apiextensionsv1.CustomResourceDefinition) (
 			crd = aCRD
 		} else {
 			if !equality.Semantic.DeepEqual(crd.Spec, aCRD.Spec) {
+				klog.V(10).Infof("Found multiple CRDs with the same name %q, but different specs: %v", crdName, cmp.Diff(crd.Spec, aCRD.Spec))
```
Comment (stevekuznetsov) [inline pkg/server/server.go:939]: V(10) means that when we have a rare flake, nobody will see this log, and you won't be able to debug :)

Comment (jmprusi) [reply pkg/server/server.go:939]: `klog.Errorf()` then? I didn't want it to be too spammy, but perhaps in this case it should

Comment (ncdc) [reply pkg/server/server.go:939]: Info (no v specified) or Error is fine for now. Let's put in a TODO to revisit the logging and/or level after we catch the flake, maybe.

**Comment by ncdc** [comment]
FYI I removed the "closes" comment in the description because this won't fix the issue, just help debug it.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**